### PR TITLE
Add more docs

### PR DESCRIPTION
**Pull request by lucasmoura**
## Proposed Commit Message
add more docs

Add docs explaining how to use the contracts staging environment and how auto-attach works.

## Test Steps
Verify if both docs are properly explaining the concepts

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [ ] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: docs/source/references/how_auto_attach_works.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+   encrypted json blob that contains the required information for the contract server
+   to verify that it is a valid *Pro* machine and that it is a valid product. Some
+   errors can happen when the contract is processing this data. For example, the *Pro*
+   image needs to have its product id registered on the contract backend, otherwise
+   the contract call will fail.
+
+3. After the metadata is validated, the contract server will return a token that can be used
+   to attach the machine into a pro subscription.
+
+4. If the attach process complete without failures, we disable the ubuntu-advantage
```
Comment (CalvoM) [inline docs/source/references/how_auto_attach_works.md:19]: Just a small correction `complete` -> `completes`.

File: docs/source/references/how_auto_attach_works.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+2. Fetch the cloud metadata and send it to contract server. The cloud metadata is an
+   encrypted json blob that contains the required information for the contract server
+   to verify that it is a valid *Pro* machine and that it is a valid product. Some
+   errors can happen when the contract is processing this data. For example, the *Pro*
+   image needs to have its product id registered on the contract backend, otherwise
+   the contract call will fail.
+
+3. After the metadata is validated, the contract server will return a token that can be used
+   to attach the machine into a pro subscription.
```
Comment (renanrodrigo) [inline docs/source/references/how_auto_attach_works.md:17]:
```suggestion
   to attach the machine to a pro subscription.
```

**Review by renanrodrigo — APPROVED** [review]
small nit, LGTM

File: dev-docs/howtoguides/use_staging_environment.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Use the contracts staging environment
+
+If needed, we can change the pro client to use the staging environment
+instead of the production environment. To do that we need to change
+the `contract_url` on the `/etc/ubuntu-advantage/uaclient.conf`. Change
+the value of that key to `https://contracts.staging.canonical.com`.
```
Comment (orndorffgrant) [inline dev-docs/howtoguides/use_staging_environment.md:6]: I slightly prefer the way you wrote a similar suggestion down in the auto attach reference.

```suggestion
You can change the contract server that UA Client will use by setting the
following option in your `uaclient.conf` configuration file, (by default located at
`/etc/ubuntu-advantage/uaclient.conf`).

`https://contracts.staging.canonical.com` is the staging environment of the
contract server

```yaml
contract_url: https://contracts.staging.canonical.com
```

**Comment by lucasmoura** [comment]
@orndorffgrant updated

File: dev-docs/explanations/how_auto_attach_works.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+   * **AWS**: http://169.254.169.254/latest/dynamic/instance-identity/pkcs7
+   * **Azure**: http://169.254.169.254/metadata/attested/document?api-version=2020-09-01
+   * **GCP**: http://metadata/computeMetadata/v1/instance/service-accounts/default/identity
+
+> **Note**
+> On some instances, like AWS, we can also use the IPv6 address to fetch the metadata
+
+3. Send this metadata json blob to the contract server at:
+
+   * https://contract.canonical.com/v1/clouds/CLOUD-TYPE/token"
```
Comment (orndorffgrant) [inline dev-docs/explanations/how_auto_attach_works.md:25]:
```suggestion
   * https://contract.canonical.com/v1/clouds/CLOUD-TYPE/token
```

**Review by orndorffgrant — APPROVED** [review]
This looks great! Just the smallest of nits: a copy-paste artifact I think :)

**Pull request merged**